rockcraft/project.py

```diff
@@ -175,15 +189,15 @@ def _validate_license(cls, rock_license):
 
     @pydantic.validator("title", always=True)
     @classmethod
-    def _validate_title(cls, title, values):
-        """If title is not provided, it default to the provided ROCK name."""
+    def _validate_title(cls, title: Optional[str], values: Any) -> str:
```

Comment:
```suggestion
    def _validate_title(cls, title: Optional[str], values: Mapping[str, Any]) -> str:
```

Or possibly `Mapping[str, str]`?